### ISSUE TITLE
Fingerprint unlocking works not always

### ISSUE THREAD

**Issue by Stevie00911** [HelloVolla/ubuntu-touch-beta-tests#99]
Ubuntu 16.04 (OTA16)

The fingerprint unlocking works not always.
After a reboot and often after a while of using the device only password unlocking is possible.

**Comment by TartanSpartan** [comment]
Confirmed.

I thought at first it was a problem because the finger was improperly placed on the sensor, but then I remembered that on the Meizu Pro 5 this was the case, but on the Volla it seems to be more along the lines of what @Stevie00911 is describing.

**Comment by natur-produkt** [comment]
Hi,
Is there any special steps to reproduce this issue? Or just:
1. Configure fingerprint
2. Wait for a while
3. Fingerprint doesn't work.

Thanks,
natur

**Comment by Stevie00911** [comment]
To reproduce this issue you can do a reboot.
After that you cannot unlock the screen with your fingerprint, only with your password.

Other steps of reproduction I cannot fix.
It appears very sporadic after a time of using the device, inclusive using the power button.

**Comment by TartanSpartan** [comment]
> To reproduce this issue you can do a reboot.
After that you cannot unlock the screen with your fingerprint, only with your password.

I believe that is intended behaviour, it was also the case on the Meizu Pro 5. I believe the rationale is the user should have to remember and input their password on a fresh boot before the fingerprinting is granted- else you might have situations like someone finding you asleep, using your fingerprint to unlock the phone from on a fresh boot and then having nearly full access to the phone (however this logic breaks down when you think, what about if the phone wasn't rebooted?). The other ways this can be reproduced (which actually are symptomatic of a bug, rather than intended behaviour) are indeed hard to find.

**Comment by stekalb** [comment]
As far as I can tell, the unlocking with the fingerprint sensor does not work if the device was in "deep sleep".
For example. Let the device in sleep mode / screen turned off over night (let's say for 8h) and try to unlock it with fingerprint at first use in the morning. As far as I can recall that never worked for me.

**Comment by wurzer** [comment]
Yes. You need to unlock the device once after deep sleep and then fingerprint works again

**Comment by Stevie00911** [comment]
OTA17 - stable
I have this issue more often.
For example afer screenlock (3 minutes).
Or after using the power button for closing and opening the screen.

**Comment by wurzer** [comment]
Please re-check this issue.

**Comment by Stevie00911** [comment]
Yes, this issue is still present (OTA 23).
In most cases of unlocking the screen the fingerprint sensor does not care about my finger and I have to type in my pin for unlocking.

**Comment by JamiKettunen** [comment]
Adding @erfanoabdi's words from [UBports Development telegram chat](https://t.me/ubports_devel/53242) here if they may be useful later in solving the seemingly unreliable fingerprint sensor after suspended for some time:
> i know how to fix it on android but thats because of some timeout is implemented on android framework for "week" auths 
so you just assign fingerprint id 0 as "strong"
> no idea about ubports, could be some irq dying in kernel..

**Comment by erfanoabdi** [comment]
Sorry for android comment but leaving codes here, maybe github search brought someone here: 
Related fwk code: https://github.com/LineageOS/android_frameworks_base/blob/lineage-18.1/services/core/java/com/android/server/locksettings/LockSettingsStrongAuth.java#L287

fix commit: https://github.com/LineageOS/android_device_motorola_nash/commit/51e37194236a5431d90c9479af75c023f827612a

**Comment by JamiKettunen** [comment]
https://github.com/ubports/unity8/pull/443 should apparently help with this issue as well, may be enough to close this altogether even :p

**Comment by Flohack74** [comment]
@Stevie00911 so to clear this up:
- After reboot the user is required to enter a PIN or passphrase once, this seems to be a security design also found on other systems.
- Later on the problem was that our shell did, in case of a misread print, rapidly increase the error counter instead of giving the user time to take away the finger. Also the text "try again" was not really visible, instead it jumped then to PIN/passphras unlock.
- We will try to improve the situation in OTA-24. I have a test branch already and will try it out over the next 2 weeks.

**Comment by wurzer** [comment]
What I can reproduce is, that after some longer time in sleep mode, I can't unlock the device with UT like with Volla OS. After a shorter time in sleep mode, I can wake up and unlock the device properly.  The fact, that I need to enter the PIN after boot is an expected behavior.

**Comment by Flohack74** [comment]
If anyone wants to test the improvement please do a sudo `ubports-qa install xenial_-_fpreader-awesome` in a terminal on the device and reboot.

**Comment by gbdomubpkm** [comment]
Unlocking the phone by diagram would be very practical too 🙂. If it  possible on UT, I'll open a ticket somewhere.

**Comment by Flohack74** [comment]
@JamiKettunen if this happens again can you provide the tail of the unity8.log  file, it should log number of failed attempts

**Comment by gbdomubpkm** [comment]
@j2g2rp You never had to reuse the unlock code since the installation of the PR ?

**Comment by gbdomubpkm** [comment]
@Flohack74 I installed the PR on RC 54. The installation was capricious: 66% blockage with a history of 'waiting bluetooth off' I believe, I didn't understand because i did nothing with bluetooth. I waited a long time and the installation seemed blocked to 66 %. In short, I closed the terminal, reopened and relaunched the same command line: after writing lines, a little weird, it told me that the PR was installed and i had to reboot. I prefered to turn off the Volla: impossible to turn it on again: I tried a forced start (pressing the power button for 15 seconds): the Volla came back on! Everything seems OK now.
The PR seems to have improved things, i observe. So I went back from the 4-digit code to the fingerprint sensor. I had to erase the registered fingerprints and re-register new ones so as to the fingerprint sensor can work.

**Comment by gbdomubpkm** [comment]
Despite the little bit of stress and the thought of having maybe bricked the Volla, all this during the installation, I confirm that the fingerprint sensor works perfectly now for me 🙂

**Comment by j2g2rp** [comment]
> @j2g2rp You never had to reuse the unlock code since the installation of the PR ?

@gbdomubpkm
Just one time yesterday but the behavior is different than before.

Before:
Unlock with the finger print failed and  showed the pin code unlock. (If I remember correctly It said "try again" when trying to unlock)

Now:
It says: Unlock with pin code or something like that like when you have to unlock by first time after a reboot.

About the problem during installation with qa-install I had more or less the same problem. Stucked during turning off the bluetooth. I had to cancel the installation and reboot the phone.

**Comment by Flohack74** [comment]
This is now merged in devel. Itfs fine if you installed the branch before to just update to latest devel if you want. Note that RC will take still a bit of time, as we have seen issues with OTA upgrades recently.

**Comment by mihaelmilea** [comment]
@Flohack74 Would this PR work for Openplus 5T stable channel as well? - I have the same issue: https://forums.ubports.com/topic/7994/not-able-to-unlock-with-fingerprint-sometimes

Thank you!

**Comment by mihaelmilea** [comment]
:) Didn't work - it wouldn't boot fully up to the login screen - it was stuck on the red UBports screen. I had to reinstall the OS.

**Comment by JamiKettunen** [comment]
@mihaelmilea if you're on Development (or I think even RC now/soon?) channel you should already have the change as Florian mentioned ~3 weeks ago now

**Comment by mihaelmilea** [comment]
I'm on stable.

**Comment by JamiKettunen** [comment]
For latest development/testing please don't use stable.

**Comment by gbdomubpkm** [comment]
Can fingerprint detection still deteriorate over time, after several months ? In my case, erasing a fingerprint and then saving it again seems to have solved this recent problem.

**Comment by wurzer** [comment]
If the fingerprint sensor gets damp or wet, the fingerprint sensor may temporarily or completely fail. In a developer version, individual functions may fail. However, this should not apply to a stable version.

**Comment by Stevie00911** [comment]
Volla Phone 22:
With the delivered Ubuntu 16.04 Version the Fingerprint worked as it had included the  "sudo ubports-qa install xenial_-_fpreader-awesome" solution and it worked verry well.
With the OTA Update (2022-08-18/2) the Fingerprint behaviour is as if the solution was delited again.

**Comment by Flohack74** [comment]
@Stevie00911 we did not change anything anymore on that feature, so thats really strange then. We run a new RC in the next 2 days, plz retest with this

**Comment by Stevie00911** [comment]
@Flohack74 with the RC "2022-W39" the fingerprint behaviour is well again, as if the solution "sudo ubports-qa install xenial_-_fpreader-awesome" is included.

**Comment by Flohack74** [comment]
@Stevie00911 no clue then ^^

**Comment by gbdomubpkm** [comment]
Note that very occasionally, randomly, there may be a certain slowness in the unlocking of the screen which can leads  to a reboot of lomiri. Seen this in RC 54 and today in RC 58 i got this.

**Comment by wurzer** [comment]
Please note, that it's a known issue, that the fingerprint sensor doesn't wake up the Volla Phone 22 device.
For all devices, the user needs to enter the PIN after booting the device. Only after entering the PIN, the fingerprint sensor is working. This is an expected behaviour.

**Comment by JamiKettunen** [comment]
Since a fix for the original problem (fingerprint trying to unlock too many times fast in a row making it quickly unusable on failed attempts) should have landed with OTA-24 which is rolling out to devices over the next 4 days or so I think then this issue can be closed. If there's further issues new tickets should be made for them specifically :)

**Comment by JamiKettunen** [comment]
Closing as per the above message. If you're experiencing issues with fingerprint sensor on OTA-24 please comment here or open a new issue! :)
